cranelift/codegen/src/isle_prelude.rs

```diff
@@ -734,6 +739,41 @@ macro_rules! isle_common_prelude_methods {
             }
         }
 
+        #[inline]
+        fn intcc_sle_or_ule(&mut self, cc: &IntCC) -> Option<IntCC> {
```

Comment:
I'd kind of prefer not to add more magic predicates on `IntCC`. In the just-merged PR #6095 we used bitfield representations of `IntCC` to avoid this. Maybe you can do something similar here?

Reply:
Hey, I tried to get this working and keeping an extractor interface for these, but I think I might be missing something about how extractors work. Here's the outlines of what I tried (Here's the full commit https://github.com/afonso360/wasmtime/commit/4f118c37cd2e95b6507cde18d7008ad9bd3165f1):

```
(decl pure intcc_bit (IntCC) u64)
(rule ...)

;; Returns true if the first IntCC is equal to any of the following IntCCs.
(decl pure partial intcc_any2 (IntCC IntCC IntCC) bool)
(rule ...)

(decl pure partial intcc_sle_or_ule (IntCC) IntCC)
(extractor (intcc_sle_or_ule cc)
  (if (intcc_any2 cc (IntCC.SignedLessThanOrEqual) (IntCC.UnsignedLessThanOrEqual)))
  cc)
```

Declaring `intcc_sle_or_ule` like this is invalid, but I don't really understand why? I would expect it to work. I think I'm just not understanding how extractors work.  

I tried going back to the original RFC, but that didn't really help. Is there any other documentation about this?

Reply:
Sorry I didn't get that first time around! That makes a lot more sense, thanks for explaining it! ❤️